inifile.cc

```diff
@@ -132,35 +128,31 @@ inifile_load_file(gbfile* fin, inifile_t* inifile, const char* myname)
               qPrintable(gbinipathname));
       }
 
-      sec = (inifile_section_t*) xcalloc(1, sizeof(*sec));
-
-      sec->name = xstrdup(cin);
-      QUEUE_INIT(&sec->entries);
-      ENQUEUE_TAIL(&inifile->secs, &sec->Q);
-      inifile->isecs++;
+      inifile->secs.append(inifile_section_t());
+      sec = &inifile->secs.last();
+      sec->name.reset(xstrdup(cin), xfree);
     } else {
       if (sec == nullptr) {
         fatal("%s: missing section header in '%s'.\n", myname,
               qPrintable(gbinipathname));
       }
 
-      inifile_entry_t* entry = (inifile_entry_t*) xcalloc(1, sizeof(*entry));
-      ENQUEUE_TAIL(&sec->entries, &entry->Q);
-      sec->ientries++;
+      sec->entries.append(inifile_entry_t());
```

Comment:
Would it be more natural to just construct entry, use it, and append it instead of constructing it, shoving it into the queue, and then immediately pulling the last one back out?